File: java/250330_S5_4673.java
/**
 * https://www.acmicpc.net/problem/4673
 */
import java.io.*;

class Main {
	public static void main(String[] args) throws IOException {
		try(var bw = new BufferedWriter(new OutputStreamWriter(System.out))) {
			int[] arr = new int[10001];
			for(int i = 1; i < 10000; i++) {
				int self = self(i);
				if (self <= 10000) {
					arr[self]++;
				}
			}
			for(int i = 1; i < 10001; i++) {
				if(arr[i] == 0) {
					bw.write("" + i);
					bw.newLine();
				}
			}
			bw.flush();
			bw.close();
		}
	}
	
	private static int self(int i) {
		int sum = i;
		while(i != 0) {
			sum += i % 10;
			i /= 10;
		}
		return sum;
	}
}

File: rust/src/bin/250330_S5_4673.rs
/**
 * https://www.acmicpc.net/problem/4673
 */

fn main() {
    let mut arr: [usize; 10000] = [0; 10000];
    for i in 1..9999 {
        let sf = get_selfnum(i);
        if sf < 10000 {
            arr[sf - 1] += 1;
        }
    }
    for i in 0..9999 {
        if arr[i] == 0 {
            println!("{}", i + 1);
        }
    }
}

fn get_selfnum(mut i: usize) -> usize {
    let mut sum = i;
    while i > 0 {
        sum += i % 10;
        i /= 10;
    }
    sum 
}
